File: go-ms/infrastructure/entry-points/rest/rest_controller.go
package rest

import (
	"github.com/gin-gonic/gin"
	"maocq/go-ms/domain/usecase"
	"net/http"
)

func Start(cases *usecase.CasesUseCase) {
	router := gin.Default()
	router.GET("/api/hello", func(c *gin.Context) { c.String(http.StatusOK, "Hello") })

	router.GET("/api/case-one", func(c *gin.Context) {
		latency := c.DefaultQuery("latency", "0")

		if result, err := cases.CaseOne(latency); err != nil {
			c.String(http.StatusInternalServerError, err.Error())
		} else {
			c.JSON(http.StatusOK, result)
		}
	})

	router.GET("/api/case-two", func(c *gin.Context) {
		latency := c.DefaultQuery("latency", "0")

		if result, err := cases.CaseTwo(latency); err != nil {
			c.String(http.StatusInternalServerError, err.Error())
		} else {
			c.JSON(http.StatusOK, result)
		}
	})

	router.GET("/api/case-three", func(c *gin.Context) {
		result := cases.CaseThree(1000)
		c.String(http.StatusOK, result)
	})

	router.Run("0.0.0.0:8080")
}

File: java-helidon-nima-ms/nima/src/main/java/com/maocq/helidon/nima/infrastructure/entrypoints/RestController.java
package com.maocq.helidon.nima.infrastructure.entrypoints;

import com.maocq.helidon.nima.domain.usecases.cases.CasesUseCase;
import io.helidon.common.http.Http;
import io.helidon.nima.webserver.WebServer;
import io.helidon.nima.webserver.http.HttpRouting;
import io.helidon.nima.webserver.http.ServerRequest;
import io.helidon.nima.webserver.http.ServerResponse;

public class RestController {
    private static final Http.HeaderValue SERVER = Http.Header.create(Http.Header.SERVER, "Nima");

    private final CasesUseCase cases;

    public RestController(CasesUseCase cases) {
        this.cases = cases;
    }

    public void start() {
        WebServer ws = WebServer.builder()
                .routing(this::routing)
                .start();
    }

    private void routing(HttpRouting.Builder rules) {
        rules.addFilter((chain, req, res) -> {
                    res.header(SERVER);
                    chain.proceed();
                })
                .get("/api/hello", (req, res) -> res.send("Hello"))
                .get("/api/case-one", this::caseOne)
                .get("/api/case-two", this::caseTwo)
                .get("/api/case-three", this::caseThree)
                .register("/", new RouterRest());
    }

    private void caseOne(ServerRequest req, ServerResponse res) {
        var latency = Integer.valueOf(req.query().first("latency").orElse("0"));
        String response = cases.caseOne(latency);
        res.send(response);
    }

    private void caseTwo(ServerRequest req, ServerResponse res) {
        var latency = Integer.valueOf(req.query().first("latency").orElse("0"));
        String response = cases.caseTwo(latency);
        res.send(response);
    }

    private void caseThree(ServerRequest req, ServerResponse res) {
        String response = cases.caseThree(1000);
        res.send(response);
    }
}
